cpa/tester/compiler_policies/parameters/parameter.rego

```diff
@@ -0,0 +1,13 @@
+package org
+
+policy_name["parameters"]
+
+enable_rule["assert_params"]
+
+# for this test, the mock test compiler will inject the pipeline parameters as the _compiled_ value
```

Comment:
This sounds confusing to me, as we are comparing two different things (parameters and compiled config).

Should we instead have an "expected_compiled" to compare expected and actual compiled value instead.

Reply:
I understand it is probably the only and a smart way to test both compiler option and parameter parsing.

Reply:
So, the agent doesn't have a compiler built-in to it. It accepts the compilation function, and so each test provides the compilation logic based on the input and pipeline parameters. 

This test just wants to prove that pipeline parameters are indeed passed and merged as expected.

To test this, the "compiler" just forwards the pipeline parameters to `_compiled_` and we can make assertions using a rule that the expected_parameters match the _compiled_ parameters.

Does that make sense?